utils/filter_streams.h

```diff
@@ -109,6 +110,25 @@ class filter_ostream : public std::ostream {
     
     void close();
 
+    /* The next four overrides only take care of handling the deferred
+       exception flag, when an exception occurs. They otherwise execute their
+       std::ostream equivalent. */
+    template<typename T> filter_ostream & operator << (T && data)
+    {
+        try {
+            std::ostream & stdStream(*this);
+            stdStream << std::forward<T>(data);
```

Comment:
I want to make sure I invoke the version from std::stream. *this is a ML::filter_ostream, which makes me think it would call the ML::filter_ostream version recursively. Now, this is template code so you tell me...